File: update_form.py
import os
import re

file_path = 'frontend/components/AddMemberForm.tsx'
with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

import_statement = """import PersonalInfoTab from './PersonalInfoTab';
import AddressTab from './AddressTab';
import FamilyTab from './FamilyTab';
import DocumentsTab from './DocumentsTab';
import ReviewTab from './ReviewTab';
"""
content = content.replace("import ConfirmModal from './ConfirmModal';", import_statement + "import ConfirmModal from './ConfirmModal';")

content = re.sub(r'\{\/\* STEP 1: IDENTITY \*\/.*?\{\/\* STEP 2: LOCATION \*\/\}',
                 '{/* STEP 1: IDENTITY */}\n          {step === 1 && (\n            <PersonalInfoTab\n              formData={formData}\n              setFormData={setFormData}\n              applicantType={applicantType}\n              isSeniorRole={isSeniorRole}\n              isMatchFound={isMatchFound}\n              lookupLoading={lookupLoading}\n            />\n          )}\n\n          {/* STEP 2: LOCATION */}', content, flags=re.DOTALL)

content = re.sub(r'\{\/\* STEP 2: LOCATION \*\/.*?\{\/\* STEP 3: FAMILY \*\/\}',
                 '{/* STEP 2: LOCATION */}\n          {step === 2 && (\n            <AddressTab\n              formData={formData}\n              setFormData={setFormData}\n            />\n          )}\n\n          {/* STEP 3: FAMILY */}', content, flags=re.DOTALL)

content = re.sub(r'\{\/\* STEP 3: FAMILY \*\/.*?\{\/\* STEP 4: DOCS \*\/\}',
                 '{/* STEP 3: FAMILY */}\n          {step === 3 && (\n            <FamilyTab\n              formData={formData}\n              setFormData={setFormData}\n              tempMember={tempMember}\n              setTempMember={setTempMember}\n              addFamilyMember={addFamilyMember}\n              removeFamilyMember={removeFamilyMember}\n            />\n          )}\n\n          {/* STEP 4: DOCS */}', content, flags=re.DOTALL)

content = re.sub(r'\{\/\* STEP 4: DOCS \*\/.*?\{\/\* STEP 5: REVIEW - Same as before \*\/\}',
                 '{/* STEP 4: DOCS */}\n          {step === 4 && (\n            <DocumentsTab\n              formData={formData}\n              setFormData={setFormData}\n              applicantType={applicantType}\n              uploadedFiles={uploadedFiles}\n              handleFileChange={handleFileChange}\n              showPassword={showPassword}\n              setShowPassword={setShowPassword}\n              showConfirmPassword={showConfirmPassword}\n              setShowConfirmPassword={setShowConfirmPassword}\n              passwordHasMin={passwordHasMin}\n              passwordHasNumber={passwordHasNumber}\n              passwordHasSpecial={passwordHasSpecial}\n            />\n          )}\n\n          {/* STEP 5: REVIEW - Same as before */}', content, flags=re.DOTALL)

content = re.sub(r'\{\/\* STEP 5: REVIEW - Same as before \*\/.*?<div className="flex flex-col sm:flex-row',
                 '{/* STEP 5: REVIEW - Same as before */}\n          {step === 5 && (\n            <ReviewTab formData={formData} applicantType={applicantType} />\n          )}\n\n          <div className="flex flex-col sm:flex-row', content, flags=re.DOTALL)

content = re.sub(r'const ReviewField = \(\{.*?\}\)\s*=>\s*\([\s\S]*?\n  \);\n', '', content)
content = re.sub(r'const FileUploadField = \(\{.*?\}\)\s*=>\s*\([\s\S]*?\n    </div>\n  \);\n', '', content)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content)

print("Updated AddMemberForm.tsx successfully.")

File: update_form.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'frontend/components/AddMemberForm.tsx');
let content = fs.readFileSync(filePath, 'utf-8');

const importStatement = `import PersonalInfoTab from './PersonalInfoTab';
import AddressTab from './AddressTab';
import FamilyTab from './FamilyTab';
import DocumentsTab from './DocumentsTab';
import ReviewTab from './ReviewTab';
`;
content = content.replace("import ConfirmModal from './ConfirmModal';", importStatement + "import ConfirmModal from './ConfirmModal';");

content = content.replace(/\{\/\* STEP 1: IDENTITY \*\/[\s\S]*?\{\/\* STEP 2: LOCATION \*\/\}/g,
                 `{/* STEP 1: IDENTITY */}
          {step === 1 && (
            <PersonalInfoTab
              formData={formData}
              setFormData={setFormData}
              applicantType={applicantType}
              isSeniorRole={isSeniorRole}
              isMatchFound={isMatchFound}
              lookupLoading={lookupLoading}
            />
          )}

          {/* STEP 2: LOCATION */}`);

content = content.replace(/\{\/\* STEP 2: LOCATION \*\/[\s\S]*?\{\/\* STEP 3: FAMILY \*\/\}/g,
                 `{/* STEP 2: LOCATION */}
          {step === 2 && (
            <AddressTab
              formData={formData}
              setFormData={setFormData}
            />
          )}

          {/* STEP 3: FAMILY */}`);

content = content.replace(/\{\/\* STEP 3: FAMILY \*\/[\s\S]*?\{\/\* STEP 4: DOCS \*\/\}/g,
                 `{/* STEP 3: FAMILY */}
          {step === 3 && (
            <FamilyTab
              formData={formData}
              setFormData={setFormData}
              tempMember={tempMember}
              setTempMember={setTempMember}
              addFamilyMember={addFamilyMember}
              removeFamilyMember={removeFamilyMember}
            />
          )}

          {/* STEP 4: DOCS */}`);

content = content.replace(/\{\/\* STEP 4: DOCS \*\/[\s\S]*?\{\/\* STEP 5: REVIEW - Same as before \*\/\}/g,
                 `{/* STEP 4: DOCS */}
          {step === 4 && (
            <DocumentsTab
              formData={formData}
              setFormData={setFormData}
              applicantType={applicantType}
              uploadedFiles={uploadedFiles}
              handleFileChange={handleFileChange}
              showPassword={showPassword}
              setShowPassword={setShowPassword}
              showConfirmPassword={showConfirmPassword}
              setShowConfirmPassword={setShowConfirmPassword}
              passwordHasMin={passwordHasMin}
              passwordHasNumber={passwordHasNumber}
              passwordHasSpecial={passwordHasSpecial}
            />
          )}

          {/* STEP 5: REVIEW - Same as before */}`);

content = content.replace(/\{\/\* STEP 5: REVIEW - Same as before \*\/[\s\S]*?<div className="flex flex-col sm:flex-row/g,
                 `{/* STEP 5: REVIEW - Same as before */}
          {step === 5 && (
            <ReviewTab formData={formData} applicantType={applicantType} />
          )}

          <div className="flex flex-col sm:flex-row`);

content = content.replace(/const ReviewField = \(\{[\s\S]*?\n  \);\n/, '');
content = content.replace(/const FileUploadField = \(\{[\s\S]*?\n    <\/div>\n  \);\n/, '');

fs.writeFileSync(filePath, content, 'utf-8');
console.log("Updated AddMemberForm.tsx using node.");
